src/lib.rs

```diff
@@ -55,7 +61,41 @@ where
             Error::Spi(spi) => write!(f, "Error::Spi({:?})", spi),
             Error::Gpio(gpio) => write!(f, "Error::Gpio({:?})", gpio),
             Error::UnexpectedStatus => f.write_str("Error::UnexpectedStatus"),
+            Error::SectorLength => f.write_str("Error::SectorLength"),
             Error::__NonExhaustive(_) => unreachable!(),
         }
     }
 }
+
+/// A trait for reading operations from a memory chip.
+pub trait Read<Addr, SPI: Transfer<u8>, CS: OutputPin> {
+    /// Reads bytes from a memory chip.
+    ///
+    /// # Parameters
+    /// * `addr`: The address to start reading at.
+    /// * `buf`: The buffer to read buf.len() bytes into.
+    fn read(&mut self, addr: Addr, buf: &mut [u8]) -> Result<(), Error<SPI, CS>>;
+}
+
+/// A trait for writing and erasing operations on a memory chip.
+pub trait BlockDevice<Addr: TryInto<usize> + Copy, SPI: Transfer<u8>, CS: OutputPin>
+where
+    <Addr as core::convert::TryInto<usize>>::Error: core::fmt::Debug,
+{
+    /// Erases sectors from the memory chip.
+    ///
+    /// # Parameters
+    /// * `addr`: The address to start erasing at.
+    fn erase_sectors(&mut self, addr: Addr, amount: usize) -> Result<(), Error<SPI, CS>>;
```

Comment:
This should document the behavior when `addr` is not on a sector boundary. I believe essentially all chips should ignore the low address bits in that case.